src/main/java/carsharing/dao/impl/CustomerDao.java

```diff
@@ -0,0 +1,88 @@
+package carsharing.dao.impl;
+
+import carsharing.Constants;
+import carsharing.connection.DataSource;
+import carsharing.model.Customer;
+import carsharing.dao.ICustomerDao;
+
+import java.sql.PreparedStatement;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CustomerDao implements ICustomerDao {
+    @Override
+    public Customer getById(int id) {
```

Comment:
You have common methods like getById(), getAll(), save() in all dao classes. Please consider to have abstract dao class with those common methods

Reply:
Done

Reply:
Done